src/transformers/tokenization_utils.py

```diff
@@ -935,17 +979,25 @@ def _decode(
     ) -> str:
         self._decode_use_source_tokenizer = kwargs.pop("use_source_tokenizer", False)
 
+        if not spaces_between_special_tokens:
```

Comment:
Based on the warning message, I would expect this to flag when `True`

```suggestion
        if spaces_between_special_tokens:
```